logging/logging.go

```diff
@@ -0,0 +1,48 @@
+package logging
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"runtime/pprof"
+)
+
+type Labels map[string]any
+
+const (
+	// FnKey refers to the function name.
+	FnKey = "fn"
+
+	// FileKey refers to the file name.
+	FileKey = "file"
+
+	// LineKey refers to the line number.
+	LineKey = "line"
+)
+
+func GoAnnotated(ctx context.Context, fn func(context.Context), labelMap ...Labels) {
```

Comment:
Since we have this nicely encapsulated, do you think you could add a build mode where all the pprof stuff is not included?

Reply:
Will do, yes

Reply:
done, annotation is disabled if `gluon_pprof_disabled` is set